src/products/Product.java

```diff
@@ -0,0 +1,127 @@
+package products;
+
+import enums.KindOfProduct;
+import enums.Models;
```

Comment:
Io metterei model e producers  come string, non come enum, come enum in caso di aggiunta di un nuovo prodotto a magazzino hai bisogno di andare a modificare il codice backend.